docs/how_to_use_clusterctl.md

```diff
@@ -0,0 +1,52 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [IBM Cloud](#ibm-cloud)
+  - [How to use clusterctl image in existing kubernetes cluster?](#how-to-use-clusterctl-image-in-existing-kubernetes-cluster)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# IBM Cloud
+## How to use clusterctl image in existing kubernetes cluster?
+The `clusterctl` image is designed to run independently to provision ibmcloud cluster. we have embedded the `kind` and `kubectl` into `clusterctl` image.
+
+1. Follow [this readme](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/#cluster-creation) to generate `cluster.yaml`, `machines.yaml`, `provider-components.yaml`, and `addons.yaml`
```

Comment:
`addon.yaml` is optional. But it seems there is no description for this and your job yaml file make it mandatory. We should clarify this.

Reply:
I think `addons.yaml` is optional  and it was declared in README file.

Reply:
For yaml file, I mean `examples/clusterctl/job.yaml`.
```       - command: ['sh', '-c', 'clusterctl create cluster --bootstrap-type kind --provider ibmcloud -c /examples/cluster.yaml -m /examples/machines.yaml -p /examples/provider-components.yaml -a /examples/addons.yaml && kubectl create secret generic kubeconfig --from-file kubeconfig']

Reply:
I am not sure where should be a right place to add such notice to notify the user. Maybe remove the `addons.yaml` from `examples/clusterctl/job.yaml`. and tell user that need manually modify the `job` and `configmap` if you want to add `addons.yaml`. WDYT?

Reply:
I think it is fine. Only use mandatory should be ok for most of the case. User who want to use optional ones should do it by themselves.

Reply:
Done.